lib/swoosh/mailer.ex

```diff
@@ -98,26 +110,26 @@ defmodule Swoosh.Mailer do
   @doc """
   Parses the OTP configuration at compile time.
   """
-  def parse_config(mailer, opts) do
-    otp_app = Keyword.fetch!(opts, :otp_app)
+  def parse_static_config(mailer, module_config) do
+    otp_app = Keyword.fetch!(module_config, :otp_app)
     config = Application.get_env(otp_app, mailer, [])
-    adapter = opts[:adapter] || config[:adapter]
```

Comment:
I don't think we should do this here anymore (getting the adapter module and raising an error if it's not set), it's something that should happen at runtime, in `deliver/2` or whatever function we use for parsing/merging config.